sdks/cpp/lite/include/Param.h

```diff
@@ -126,10 +140,11 @@ template <typename T> class Param : public IParam {
       }
     }
 
-  private:
+private:
     ParamType type_;  // ParamType is from param.pb.h
     std::vector<std::string> oid_aliases_;
     PolyglotText name_;
+    catena::common::IConstraint* constraint_;
```

Comment:
this would simplify to `IConstraint* constraint_;` by using that typedef I mentioned earlier.